global/scripts/CorrThick.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+set -eu
+
+pipedirguessed=0
+if [[ ""$HCPPIPEDIR:-"" == "" ]]
+then
+	pipedirguessed=1
+	#fix this if the script is more than one level below HCPPIPEDIR
+	export HCPPIPEDIR="$(dirname -- "$0")/.."
+fi
+
+source "$HCPPIPEDIR/global/scripts/newopts.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/debug.shlib" "$@"
+
+#description of script/command
+opts_SetScriptDescription "Run curvature-corrected cortical thickness python script and save curvatures, regression coefficients, and curvature-corrected thickness"
+
+opts_AddMandatory '--subject-dir' 'SubjectDir' 'path' "folder containing all subjects"
+opts_AddMandatory '--subject' 'Subject' 'subject ID' "subject-id"
+opts_AddOptional '--regnamestr' 'RegNamesStr' 'my reg str' "set the desired registration name(s) separated by @, 'string' 'RegName@RegName@RegName@...etc.' default MSMSulc" "MSMSulc"
+opts_AddOptional '--hemi' 'Hemi' 'hemisphere' "provide hemisphere for regression calculation, L=Left, R=Right, or B=Both, default 'B'" "B"
+opts_AddOptional '--surf' 'Surface' 'surface' "provide surface for regression calculation, white or midthickness, default midthickness" "midthickness"
+opts_AddOptional '--patch-size' 'PatchSize' 'distance' "provide patch kernel size in millimeters FWHM for regression, default 6" "6"
+opts_AddOptional '--surf-smooth' 'SurfSmooth' 'distance' "provide surface smoothing in millimeters FWHM, default 2.14" "2.14"
+opts_AddOptional '--metric-smooth' 'MetricSmooth' 'distance' "provide metric smoothing in millimeters FWHM, default 2.52" "2.52"
+opts_AddOptional '--resample-only' 'ResampleOnly' 'YES or NO' "whether or not to generate the curvature-corrected cortical thickness data, if it is already available, but just to resample it to 164k and 32k, defaults to 'NO'" "NO"
+
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+	log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+#display the parsed/default values
+opts_ShowValues
+
+#set paths
+NonlinearFolder="$SubjectDir"/"$Subject"/MNINonLinear
+NativeFolder="$NonlinearFolder"/Native
+T1wFolder="$SubjectDir"/"$Subject"/T1w/Native
+
+#Make intermediate directory to save intermediate files
+mkdir -p "$NativeFolder"/CorrThick
+
+#Loop through left and right hemispheres
+if [[ "$Hemi" == *B* ]]
+then
+	Hemi="L R"
+fi
+
+RegNames=`echo "$RegNamesStr" | sed s/"@"/" "/g`
+
+LowResMesh="32"
+HighResMesh="164"
+MapListFunc="MRcorrThickness MRcorrThickness_intercept MRcorrThickness_normcoeffs MRcorrThickness_curvs MRcorrThickness_coeffs"
+
+#Generate MRcorrThickness in Native Space
+if [[ "$ResampleOnly" == "NO" ]]; then
```

Comment:
Please use `opts_StringToBool` to sanity check yes/no type arguments (converts them to 0/1, which can also be used in math parenthesis for a neater-looking conditional):

https://github.com/Washington-University/HCPpipelines/blob/b4f33c5df48b0f0402f9685b7f35fd7c58ce4720/global/scripts/RSNregression.sh#L72
https://github.com/Washington-University/HCPpipelines/blob/b4f33c5df48b0f0402f9685b7f35fd7c58ce4720/global/scripts/RSNregression.sh#L211

Reply:
To be clear though, in order to do "not <flag>" with math parentheses, you need to do this:

```
if ((! DoZ))
```
The space between the `!` and the variable name is required, otherwise bash thinks it is a history expansion.